tests/util/caches/test_response_cache.py

```diff
@@ -129,3 +143,49 @@ def test_cache_wait_expire(self):
         self.reactor.pump((2,))
 
         self.assertIsNone(cache.get(0), "cache should not have the result now")
+
+    @parameterized.expand([(True,), (False,)])
+    def test_cache_context_nocache(self, should_cache: bool):
+        """If the callback clears the should_cache bit, the result should not be cached"""
+        cache = self.with_cache("medium_cache", ms=3000)
+
+        expected_result = "howdy"
+
+        call_count = [0]
+
+        async def non_caching(o: str, cache_context: ResponseCacheContext[int]):
+            call_count[0] += 1
```

Comment:
I understand what is happening here, the actual variable needs to be referenced, and that is done by putting it inside a list.

You can make nonlocal variable names mutable by using `nonlocal`, so this can be the following instead (also changing above to `call_count = 0`, and below to `self.assertEqual(call_count, 1)`);

```suggestion
            nonlocal call_count
            call_count += 1
```

Reply:
TIL

Reply:
FWIW the same is possible with `global`